### PR TITLE
Maryia/DTRA-1441/feat: add last digit selector and expand displayed spot feature for digits

### PR DESCRIPTION
**Pull request by maryia-deriv**
## Changes:

- add last digit selector 
- add current spot display for ongoing digit contracts

*animation not included

### CODE REVIEW

File: packages/trader/src/AppV2/Containers/Chart/trade-chart.tsx [hunk 1]
```diff
@@ -8,6 +8,30 @@ import { SmartChart } from 'Modules/SmartChart';
...
+    }, [tick]);
+
+    React.useEffect(() => {
+        setDigitStats(digits);
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [digits]);
+
+    // render no bottom widgets on chart
+    return null;
+};
```
Comment (maryia-deriv) [inline packages/trader/src/AppV2/Containers/Chart/trade-chart.tsx:33]: applied the same logic as we have in production

File: packages/trader/src/Stores/Modules/Trading/trade-store.ts [hunk 2]
```diff
@@ -1686,41 +1690,32 @@ export default class TradeStore extends BaseStore {
...
-            const symbol = response.echo_req.ticks_history;
-            if (this.symbol !== symbol) return;
-            this.tick_data = {
-                current_spot: prices?.[prices?.length - 1],
-                current_spot_time: times?.[times?.length - 1],
-                prev_spot_time: times?.[times?.length - 2],
-                pip_size: response.pip_size as number,
-            };
-        }
-    }
```
Comment (maryia-deriv) [inline packages/trader/src/Stores/Modules/Trading/trade-store.ts:1713]: removing new observables and actions which are no longer needed

File: packages/trader/src/Stores/Modules/Trading/trade-store.ts [hunk 3]
```diff
@@ -1686,41 +1690,32 @@ export default class TradeStore extends BaseStore {
...
+                    if (this.symbol !== symbol) return;
+                    current_spot_data = {
+                        current_spot: prices?.[prices?.length - 1],
+                        current_spot_time: times?.[times?.length - 1],
+                        prev_spot_time: times?.[times?.length - 2],
+                    };
+                } else {
+                    return;
                 }
+                this.root_store.contract_trade.updateAccumulatorBarriersData(current_spot_data);
```
Comment (maryia-deriv) [inline packages/trader/src/Stores/Modules/Trading/trade-store.ts:1718]: restoring production code

File: packages/trader/src/AppV2/Components/CurrentSpot/current-spot.tsx [hunk 4]
```diff
@@ -1,35 +1,131 @@
...
+        } = {
+            DIGITMATCH: (val: number | null) => val === barrier,
+            DIGITDIFF: (val: number | null) => val !== barrier,
+            DIGITOVER: (val: number | null) => !!((val || val === 0) && (barrier || barrier === 0)) && val > barrier,
+            DIGITUNDER: (val: number | null) => !!((val || val === 0) && (barrier || barrier === 0)) && val < barrier,
+            DIGITODD: (val: number | null) => !!val && Boolean(val % 2),
+            DIGITEVEN: (val: number | null) => (!!val && !(val % 2)) || val === 0,
+        };
+        if (!contract_type || !barrier_map[contract_type]) return null;
+        return barrier_map[contract_type](num) ? num : null;
```
Comment (maryia-deriv) [inline packages/trader/src/AppV2/Components/CurrentSpot/current-spot.tsx:81]: used logic from production

**Review by kate-deriv — APPROVED** [review]

**Pull request merged**